components/element/breakpointToggle/src/viewport-resize.js

```diff
@@ -0,0 +1,62 @@
+import throttle from 'lodash.throttle'
+
+/**
+ * Throttle time for events' listeners
+ * @type {Number}
+ */
+const THROTTLE_TIME = 500
+
+/**
+ * Options for resize event's listeners
+ * @type {Object}
+ */
+const EVENT_OPTIONS = {capture: true, passive: true}
+
+/**
+ * Callbacks to execute on each resize event.
+ * @type {Array<Function>}
+ */
+const listeners = []
+
+/**
+ * Service to get width and height of the viewport
+ */
+class ViewportResize {
+  /**
+   * Register event to update viewport width and height
+   * @param {Function} listener
+   */
+  static addListener (listener) {
+    listeners.push(listener)
+    this.update()
+
+    if (listeners.length) {
+      window.addEventListener('resize', this.update, EVENT_OPTIONS)
+    }
+  }
+
+  /**
+   * Unregister event to update viewport width and height
+   * @param {Function} listener
+   */
+  static removeListener (listener) {
+    listeners.filter(cb => cb !== listener)
```

Comment:
I dont get this line.

Reply:
It's a mistake. It should be
```js
listeners = listeners.filter(cb => cb !== listener)
```

Can you do it @davidmartin84 ?
Actually once we move it somwhere, we must test it.

Reply:
Done!